java/013-strstr.md

```diff
@@ -4,9 +4,32 @@
 - [next: 014. First Position of Target](014-first-position-of-target.md)
 
 ---
-
-put your own notes and solutions here.
-you can add any reference link such as [title](reference url) here.
+###Note:
+[substring](http://www.tutorialspoint.com/java/java_string_substring.htm)
+###Solution:
+```java
+class Solution {
+    /**
+     * Returns a index to the first occurrence of target in source,
+     * or -1  if target is not part of source.
+     * @param source string to be scanned.
+     * @param target string containing the sequence of characters to match.
+     */
+    public int strStr(String source, String target) {
+        if(source == null || target == null || target.length()>source.length()) 
+            return -1;
+        for(int i=0;i<source.length()-target.length()+1;i++){
+            // substring(beg_ind, end_ind), includes beg_ind, excludes end_ind
+            String substr = source.substring(i,i+target.length());
```

Comment:
Hello there, I think there would be a little room of improvement of your code. Maybe here using String.charAt() to check the equality of each character between source and target could be better. String.substring() would create a new copy of string, which would be space expensive. e.g. worst case we might have O(mn) space complexity here ( m is source.length and n is target.length).